GetBonusActivation/handler.ts

```diff
@@ -54,46 +76,72 @@ export function GetBonusActivationHandler(
             new Error(`Query Error code=${queryError.code}|${queryError.body}`)
         )
       )
-      .fold<IGetBonusActivationHandlerOutput>(
+      .foldTask<IGetBonusActivationHandlerOutput>(
         err => {
           const error = `GetBonusActivation|ERROR|Error: [${err.message}]`;
           context.log.error(error);
-          return ResponseErrorInternal(error);
+          return new Task(async () => ResponseErrorInternal(error));
         },
         maybeBonusActivation => {
           if (isSome(maybeBonusActivation)) {
-            return toApiBonusActivation(maybeBonusActivation.value).fold<
-              IResponseSuccessJson<BonusActivation> | IResponseErrorInternal
+            return new Task(async () =>
+              toApiBonusActivation(maybeBonusActivation.value).fold<
+                IResponseSuccessJson<BonusActivation> | IResponseErrorInternal
+              >(
+                err => {
+                  const error = `GetBonusActivation|ERROR|Conversion Error: [${readableReport(
+                    err
+                  )}]`;
+                  context.log.error(error);
+                  return ResponseErrorInternal(error);
+                },
+                bonusActivation => ResponseSuccessJson(bonusActivation)
+              )
+            );
+          }
+
+          return getBonusProcessing(bonusProcessingModel, fiscalCode)
+            .foldTaskEither<
+              IResponseErrorInternal | IResponseSuccessAccepted<InstanceId>,
+              Option<BonusProcessing>
             >(
               err => {
-                const error = `GetBonusActivation|ERROR|Conversion Error: [${readableReport(
-                  err
-                )}]`;
-                context.log.error(error);
-                return ResponseErrorInternal(error);
+                context.log.warn(
+                  `StartBonusActivationHandler|WARN|Failed reading BonusProcessing|ERR=${JSON.stringify(
+                    err
+                  )}`
+                );
+                return fromLeft(
+                  ResponseErrorInternal("Failed reading BonusProcessing")
+                );
               },
-              bonusActivation => ResponseSuccessJson(bonusActivation)
+              _ => taskEither.of(_)
+            )
```

Comment:
I think just `mapLeft` is enough
```suggestion
            .mapLeft<IResponseErrorInternal | IResponseSuccessAccepted<InstanceId>>(err => {
                 context.log.warn( 
                     `StartBonusActivationHandler|WARN|Failed reading BonusProcessing|ERR=${JSON.stringify(err)}`
                  );         
                 return ResponseErrorInternal("Failed reading BonusProcessing");
            )
```